src/VerifiablePresentationManager.ts

```diff
@@ -22,18 +22,16 @@ import { PresentationVerifier, VerifyFunction } from './PresentationVerifier';
 const NESTED_PATH_DELIMITER = '.';
 const getFlattenedPaths = (objToMatch: { [prop: string]: any }, paths: any[], pathPrepend: string = '') => {
     let localPaths: any[] = paths;
-    R.keys(objToMatch).forEach((key: string) => {
-        const field = objToMatch[key];
+    R.forEachObjIndexed((value: any, key: string) => {
         const currentPath = `${pathPrepend}${(pathPrepend ? NESTED_PATH_DELIMITER : '')}${key}`;
-        // check if the type is an Object, if it is we must recurse, otherwise we don't
-        if (!['object', 'array'].includes(typeof field)) {
-            localPaths.push(currentPath);
+        // recurse objects and arrays, otherwise just push to output array
+        if (['object', 'array'].includes(typeof value)) {
+            localPaths = getFlattenedPaths(value, localPaths, currentPath);
```

Comment:
should this be pushAll? It seems like if there are two keys in objToMatch, a and b, and a is a string, so gets added to localPaths, b is an array so getflattenedpaths gets called recursively, then the result overwrites localPaths, removing a. Is that intentional?